src/components/@shared/UI/Input.tsx

```diff
@@ -0,0 +1,70 @@
+import { ReactNode } from "react";
+import { twMerge } from "tailwind-merge";
+
+interface InputProps {
+  type?: string; // 입력 필드 타입
+  placeholder?: string; // placeholder
+  buttonContent?: ReactNode; // 버튼에 들어갈 이지미, 텍스트
+  errorMessage?: string; // 에러 메시지
+  isValid?: boolean; // 유효성 검사
+  inputClassName?: string; // input 스타일
+  buttonClassName?: string; // 버튼 스타일
+  value?: string;
+  onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void; // 입력 값 변경 핸들러
+  onButtonClick?: () => void; // 버튼 클릭 핸들러
+}
+
+export default function Input({
+  type = "text",
+  placeholder = "",
+  buttonContent = null,
+  errorMessage = "",
+  isValid = true,
+  inputClassName = "",
+  buttonClassName = "",
+  value = "",
+  onChange,
+  onButtonClick,
+}: InputProps) {
+  const inputClasses = twMerge(
+    `h-[48px] w-[460px] rounded-xl border-border-primary border-opacity-10 bg-background-secondary p-4 placeholder:text-text-default hover:border-interaction-hover focus:border-interaction-hover focus:outline-none focus:ring-0 ${isValid ? "border-border-primary" : "border-status-danger focus:border-status-danger"}`,
+    inputClassName,
+  );
```

Comment:
InputProps가 InputHTMLAttributes를 extends 하도록 수정하면 여기서도 프롭들을 일일이 정의하는 대신 스프레드 ...props 로 input에 들어갈 수 있는 모든 프롭들을 받을 수 있습니다.
```suggestion
export default function Input({
  buttonContent = null,
  errorMessage = "",
  isValid = true,
  className,
  buttonClassName = "",
  onButtonClick,
  ...props,
}: InputProps) {
  const inputClasses = twMerge(
    `h-[48px] w-[460px] rounded-xl border-border-primary border-opacity-10 bg-background-secondary p-4 placeholder:text-text-default hover:border-interaction-hover focus:border-interaction-hover focus:outline-none focus:ring-0 ${isValid ? "border-border-primary" : "border-status-danger focus:border-status-danger"}`,
    className,
  );
```